src/ducks/upload/UploadQueue.jsx

```diff
@@ -48,20 +48,36 @@ class UploadQueue extends Component {
   }
 
   render () {
-    const { t, queue, doneCount } = this.props
+    const { t, queue, doneCount, successCount, purgeQueue } = this.props
     const { collapsed } = this.state
     return (
       <div className={classNames(styles['upload-queue'], {
         [styles['upload-queue--visible']]: queue.length !== 0,
         [styles['upload-queue--collapsed']]: collapsed
       })}>
         <h4 className={styles['upload-queue-header']} onDoubleClick={this.toggleCollapsed}>
-          <span className='coz-desktop'>
-            {t('UploadQueue.header', { smart_count: queue.length })}
-          </span>
-          <span className='coz-mobile'>
-            {t('UploadQueue.header_mobile', { done: doneCount, total: queue.length })}
-          </span>
+          {
+            doneCount < queue.length &&
+            <div className={styles['upload-queue-header-inner']}>
+              <span className='coz-desktop'>
+                {t('UploadQueue.header', { smart_count: queue.length })}
+              </span>
+              <span className='coz-mobile'>
+                {t('UploadQueue.header_mobile', { done: doneCount, total: queue.length })}
+              </span>
+            </div>
+          }
+          {
+            doneCount >= queue.length &&
+            <div className={styles['upload-queue-header-inner']}>
+              <span>
+                {t('UploadQueue.header_done', { done: successCount, total: queue.length })}
+              </span>
+              <button className={classNames(styles['btn-close'])} onClick={purgeQueue}>
+                { t('UploadQueue.close') }
+              </button>
+            </div>
+          }
```

Comment:
It seems both block are sharing a lot of things.
You may factorize them, maybe with an HOC or a wrapping component.